--- origsrc/glib-2.66.0/glib/gspawn.c	2020-09-10 19:42:41.605684800 +0900
+++ src/glib-2.66.0/glib/gspawn.c	2020-09-27 18:10:41.166244000 +0900
@@ -1273,7 +1273,7 @@
   if (getrlimit (RLIMIT_NOFILE, &rl) == 0 && rl.rlim_max != RLIM_INFINITY)
     open_max = rl.rlim_max;
 #endif
-#if defined(__FreeBSD__) || defined(__OpenBSD__)
+#if defined(__FreeBSD__) || defined(__OpenBSD__) || defined (__CYGWIN__)
   /* Use sysconf() function provided by the system if it is known to be
    * async-signal safe.
    *
